bootstore/src/schemes/v0/peer.rs

```diff
@@ -850,11 +893,82 @@ mod tests {
         learner_jh.await.unwrap();
         handle0.load_rack_secret().await.unwrap_err();
 
-        // TODO: Once we have persistence, we can bring an old node back from the dead
-        // and reload the rack secret.
+        // Reload an node from persistent state and successfully reload the
+        // rack secret.
+        let (mut node1, handle1) = Node::new(config[1].clone(), &log).await;
+        let jh1 = tokio::spawn(async move {
+            node1.run().await;
+        });
+        let _ = handle1.load_peer_addresses(addrs.clone()).await;
+        handle0.load_rack_secret().await.unwrap();
+
+        // Add a second learner and ensure that the generation gets bumped for
+        // either node0 or node1
+        let peer0_gen =
+            handle0.get_status().await.unwrap().fsm_ledger_generation;
+        let peer1_gen =
+            handle1.get_status().await.unwrap().fsm_ledger_generation;
+        let learner_config = learner_config(&tempdir, 2);
+        let (mut learner, learner_handle) =
+            Node::new(learner_config.clone(), &log).await;
+        let learner_jh = tokio::spawn(async move {
+            learner.run().await;
+        });
+
+        // Inform the learner, node0, and node1 about all addresses including
+        // the learner. This simulates DDM discovery
+        addrs.insert(learner_config.addr.clone());
+        let _ = learner_handle.load_peer_addresses(addrs.clone()).await;
+        let _ = handle0.load_peer_addresses(addrs.clone()).await;
+        let _ = handle1.load_peer_addresses(addrs.clone()).await;
 
-        // Shutdown node0
+        // Tell the learner to go ahead and learn its share.
+        learner_handle.init_learner().await.unwrap();
+
+        // Get the new generation numbers
+        let peer0_gen_new =
+            handle0.get_status().await.unwrap().fsm_ledger_generation;
+        let peer1_gen_new =
+            handle1.get_status().await.unwrap().fsm_ledger_generation;
+
+        // Ensure only one of the peers generation numbers gets bumped
+        assert!(
+            (peer0_gen_new == peer0_gen && peer1_gen_new == peer1_gen + 1)
+                || (peer0_gen_new == peer0_gen + 1
+                    && peer1_gen_new == peer1_gen)
+        );
+
+        // Wipe the learner ledger, restart the learner and instruct it to
+        // relearn its share, and ensure that the neither generation number gets
+        // bumped because persistence doesn't occur.
```

Comment:
This is deterministic because of learner selection of peers to learn from and the fact that the learner timeout is 5 seconds, which is an eternity in terms of computational load here.